src/states_screens/gp_info_screen.cpp

```diff
@@ -235,7 +235,7 @@ void GPInfoScreen::init()
 
         // Avoid negative numbers (which can happen if e.g. the number of karts
         // in a previous race was lower than the number of players now.
-        int num_ai = UserConfigParams::m_num_karts - race_manager->getNumLocalPlayers();
+        int num_ai = UserConfigParams::m_karts_per_gamemode[RaceManager::MAJOR_MODE_GRAND_PRIX] - race_manager->getNumLocalPlayers();
```

Comment:
it could be worth creating a new dictionary that stores per grand-prix, however that may be overkill.